goal_src/levels/beach/seagull.gc

```diff
@@ -5,3 +5,2161 @@
+   (thrust            float                   :offset-assert 224)
+   (teleport          symbol                  :offset-assert 228)
+   )
+  :heap-base #x80
+  :method-count-assert 28
+  :size-assert         #xe8
+  :flag-assert         #x1c008000e8
+  (:methods
+    (move-vertically! (_type_ symbol) none 20)
+    (adjust-heading-around-point-slow! (_type_ float) none 21)
+    (dummy-22 (_type_) none 22)
+    (adjust-heading-around-point! (_type_ float) none 23)
+    (dummy-24 (_type_) none 24)
+    (dummy-25 (_type_ float) none 25)
+    (dummy-26 (_type_) symbol 26)
+    (dummy-27 (_type_) none 27)
+    )
+  (:states
+    (seagull-flying seagull)
+    (seagull-idle seagull)
+    (seagull-landing float seagull)
+    (seagull-soaring seagull)
+    (seagull-takeoff seagull)
+    )
+  )
+
+
+(deftype seagullflock (process)
+  ((self-override      seagullflock              :offset         28)
+   (path               path-control              :offset-assert 112)
+   (trans              vector            :inline :offset-assert 128)
+   (bird               (pointer seagull) 64      :offset-assert 144)
+   (birds              int32                     :offset-assert 400)
+   (link               actor-link-info           :offset-assert 404)
+   (bird-at-waterfall  uint64                    :offset-assert 408)
+   (birds-at-waterfall int32                     :offset-assert 416)
+   (target             vector            :inline :offset-assert 432)
+   (targetnum          int32                     :offset-assert 448)
+   (alert-time         uint64                    :offset-assert 456)
+   (teleport-frames    int32                     :offset-assert 464)
+   (cam-tracker        uint64                    :offset-assert 472)
+   (state-time         uint64                    :offset-assert 480)
+   (squall             ambient-sound             :offset-assert 488)
+   (max-lift           float                     :offset-assert 492)
+   )
+  :heap-base #x180
+  :method-count-assert 17
+  :size-assert         #x1f0
+  :flag-assert         #x11018001f0
+  (:methods
+    (create-the-flock! (_type_ vector) (pointer process) 14)
+    (play-hint (_type_ int) none 15)
+    (dummy-16 (_type_ seagull) float 16)
+    )
+  (:states
+    (seagullflock-at-waterfall seagullflock)
+    (seagullflock-idle seagullflock)
+    )
+  )
+
+
+(defmethod relocate seagullflock ((obj seagullflock) (arg0 int))
+  (if (nonzero? (-> obj path))
+   (&+! (-> obj path) arg0)
+   )
+  (if (nonzero? (-> obj link))
+   (&+! (-> obj link) arg0)
+   )
+  (if (nonzero? (-> obj squall))
+   (&+! (-> obj squall) arg0)
+   )
+  (the-as seagullflock ((method-of-type process relocate) obj arg0))
+  )
+
+(defmethod deactivate seagullflock ((obj seagullflock))
+  (if (nonzero? (-> obj squall))
+   (stop! (-> obj squall))
+   )
+  ((method-of-type process deactivate) obj)
+  (none)
+  )
+
+(define
+  *seagull-boxes*
+  (new 'static 'inline-array air-box 10
```

Comment:
looks like there are 10 "zones" for the seagulls, weird. It seems like there are just 3 in the game.